aioshelly/ble/const.py

```diff
@@ -10,31 +10,63 @@
 VAR_EVENT_TYPE = "%event_type%"
 VAR_ACTIVE = "%active%"
 VAR_VERSION = "%version%"
-VAR_INTERVAL_MS = "%interval_ms%"
-VAR_WINDOW_MS = "%window_ms%"
-VAR_DURATION_MS = "%duration_ms%"
-
-DEFAULT_INTERVAL_MS = 320
-DEFAULT_WINDOW_MS = 30
-DEFAULT_DURATION_MS = -1
 
 BLE_CODE = """
-// aioshelly BLE script 1.0
-BLE.Scanner.Subscribe(function (ev, res) {
-    if (ev === BLE.Scanner.SCAN_RESULT) {
-        Shelly.emitEvent("%event_type%", [
-            %version%,
-            res.addr,
-            res.rssi,
-            btoa(res.advData),
-            btoa(res.scanRsp)
-        ]);
+// aioshelly BLE script 2.0
+const queueServeTimer = 450; // in ms, timer for events emitting
+const burstSendCount =  5; // number if events, emitted on timer event
+const maxQueue =  80; // if the queue exceeds the limit, all new events are ignored until it empties
+
+let queue = [];
+let timerHandler = null;
+
+function timerCallback() {  
+  Timer.clear(timerHandler);
+  
+  for(let i = 0; i < burstSendCount; i++) {
+    if (queue.length <= 0) {
+      break;
     }
-});
-BLE.Scanner.Start({
-    duration_ms: %duration_ms%,
+
+    Shelly.emitEvent("%event_type%", queue[0]);
+    queue = queue.slice(1);
+  }
+    
+  timerHandler = null;
+  if (queue.length > 0) {
+    timerHandler = Timer.set(queueServeTimer, false, timerCallback);
+  }
+}
+
+function bleCallback(event, res) {
+  if (event !== BLE.Scanner.SCAN_RESULT) {
+    return
+  }
+  
+  if (queue.length > maxQueue) {
+    return;
+  }
+
+  queue.push([
+    %version%,
+    res.addr,
+    res.rssi,
+    btoa(res.advData),
+    btoa(res.scanRsp)
+  ]);
+
+  if(!timerHandler) {
+    timerHandler = Timer.set(queueServeTimer, false, timerCallback);
+  }
+}
+
+// Skip starting if scanner is active
+if (!BLE.Scanner.isRunning()) {
+  BLE.Scanner.Start({
+    duration_ms: -1,
     active: %active%,
-    interval_ms: %interval_ms%,
-    window_ms: %window_ms%,
```

Comment:
Why drop the interval and window?

Reply:
I thought the default could be used, but it's better to keep it as it was. I just bring them back.